src/plugins/mer/mersdk.cpp

```diff
@@ -529,17 +529,26 @@ bool MerSdk::addTarget(const MerTarget &target)
     QScopedPointer<MerQtVersion> version(target.createQtVersion());
     if (version.isNull())
         return false;
-    std::unique_ptr<Kit> kit = target.createKit();
-    if (!kit)
+    Kit* kitptr = target.kit();
+    std::unique_ptr<Kit> kit(nullptr);
+    if (!kitptr) {
+        kit = target.createKit();
```

Comment:
It took me a while to spot the trick again now that I returned to it after some time and I would have one suggestion: I would remove MerTarget::createKit, called make_unique<Kit>() directly here, move the sysroot check to finalizeKitCreation, let it return bool and check that here.

Reply:
Thanks. This code is really ugly - let's see if it turns out any prettier with your suggestions.

Reply:
Done like you suggested. It's a little bit prettier now. The annoying pointer juggling is still there, but I'm not sure if it's possible to completely get rid of it.